server/app/assets/javascripts/file_upload.ts

```diff
@@ -39,17 +39,23 @@ export function init() {
   })
 
   const uploadedDivs = blockForm.querySelectorAll(`[${UPLOAD_ATTR}]`)
-  if (uploadedDivs.length) {
-    const uploadedDiv: HTMLDivElement = uploadedDivs[0] as HTMLDivElement
-    const uploadText = assertNotNull(uploadedDiv.getAttribute(UPLOAD_ATTR))
 
-    blockForm.addEventListener('change', (event) => {
-      const files = (event.target! as HTMLInputElement).files
-      const file = assertNotNull(files)[0]
+  blockForm.addEventListener('change', (event) => {
+    const files = (event.target! as HTMLInputElement).files
+    const file = assertNotNull(files)[0]
+    if (uploadedDivs.length) {
+      const uploadedDiv: HTMLDivElement = uploadedDivs[0] as HTMLDivElement
+      const uploadText = assertNotNull(uploadedDiv.getAttribute(UPLOAD_ATTR))
       uploadedDiv.innerText = uploadText.replace('{0}', file.name)
-      validateFileUploadQuestion(blockForm)
-    })
-  }
+    }
+
+    // If we don't have the div showing the latest file upload, the multiple file upload
+    // feature is enabled, in that case, submit the form as soon as the applicant selects
```

Comment:
Nope this is correct because this is a div from the old behavior so if it isn't there, then multiple files is enabled. but yeah this is very confusing. Do you think I should maybe pass some explicit variable instead of relying on this behavior? I didn't because we'll be removing it eventually and it's actually a bit difficult to pass down info to these files (have to embed it in some special attribute in the DOM). But happy to do so.

Reply:
I updated the comment to at least try to help.